tgui/packages/tgui/components/TimeDisplay.js

```diff
@@ -51,19 +53,32 @@ export class TimeDisplay extends Component {
     clearInterval(this.timer);
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.timing) {
+      return null;
+    }
+    return { value: nextProps.value };
+  }
+
+  defaultFormat(value) {
```

Comment:
Looks like this can either be `static` or pulled out of the component and made a util function.

Reply:
static for now, might merge into `formatTime` down the road